Source/WebCore/xml/parser/XMLDocumentParserLibxml2.cpp

```diff
@@ -867,29 +867,30 @@ void XMLDocumentParser::endElementNs()
         return;
 
     RefPtr<ContainerNode> node = m_currentNode;
-    node->finishParsingChildren();
+    auto* element = dynamicDowncast<Element>(*node);
+
+    if (element)
+        element->finishParsingChildren();
 
     if (!scriptingContentIsAllowed(parserContentPolicy()) && is<Element>(*node) && isScriptElement(downcast<Element>(*node))) {
```

Comment:
Might be nice to update this at the same time to take advantage of the new element variable.

Reply:
Good point. Will fix.